exec/internal/compile/backend_amd64.go

```diff
@@ -88,6 +128,32 @@ func (b *AMD64Backend) Build(candidate CompilationCandidate, code []byte, meta *
 	for i := candidate.StartInstruction; i < candidate.EndInstruction; i++ {
 		//fmt.Printf("i=%d, meta=%+v, len=%d\n", i, meta.Instructions[i], len(code))
 		inst := meta.Instructions[i]
+
+		// Optimization: Const followed by binary instruction: sometimes can be
+		// reduced to a single operation.
+		if (inst.Op == ops.I64Const || inst.Op == ops.I32Const) && (i+1) < candidate.EndInstruction {
+			imm := b.readIntImmediate(code, inst)
+			nextInst := meta.Instructions[i+1]
+			if _, ok := rhsConstOptimizable[nextInst.Op]; ok && imm < 256 && imm >= 0 {
```

Comment:
I would use a `switch` here:
```go
switch _, ok := rhsConstOptimizable[nextInst.Op]; {
case ok && 0 <= imm && imm < 256:
    ...
case nextInst.Op == ops.SetLocal:
    ...
}
```
(and rewrite the condition on `imm` to read more like an interval: `0 <= imm && imm < 256`)